### PR TITLE
Cache address-space-legalization of `kIROp_Store`

### PR DESCRIPTION
**Pull request by ArielG-NV**
fixes: #4479

fixes the following tests (related to #4291):
1. `tests/language-feature/types/opaque/inout-param-opaque-type.slang.1 (mtl)`
2. `tests/language-feature/types/opaque/out-param-opaque-type.slang.1 (mtl)`

problem:
Without caching a processed `kIROp_Store` will be re-processed multiple-times (or forever) which will cause an infinite-loop or crash.

solution:
Cache `kIROp_Store` processed by address-space-legalization.

Note:
Following tests now compile but still fail due to GFX backend issues (I don't have the ability to debug this to figure out what is happening):
1. `tests/language-feature/types/opaque/return-opaque-type-in-struct.slang.1 (mtl)`
2. `tests/language-feature/types/opaque/inout-param-opaque-type-in-struct.slang.1 (mtl)`
3. `tests/language-feature/types/opaque/out-param-opaque-type-in-struct.slang.1 (mtl)`

### CODE REVIEW

**Review by jkwak-work — DISMISSED** [review]
Looks good to me.

File: tests/language-feature/types/opaque/inout-param-opaque-type-in-struct.slang [hunk 1]
```diff
@@ -4,7 +4,7 @@
 // aggregate type that includes an opaque type
 
 //TEST(compute):COMPARE_COMPUTE:
-//DISABLE_TEST(compute):COMPARE_COMPUTE:-slang -shaderobj -mtl
+//TEST(compute):COMPARE_COMPUTE:-slang -shaderobj -mtl
```
Comment (jkwak-work) [inline tests/language-feature/types/opaque/inout-param-opaque-type-in-struct.slang:7]: This test appears to be failing,
```
FAILED test: 'tests/language-feature/types/opaque/inout-param-opaque-type-in-struct.slang.1 (mtl)' 
EXPECTED{{{
6070102
6080103
6090104
60A0105
}}}
ACTUAL{{{
0
0
0
0
}}}
```

File: tests/language-feature/types/opaque/out-param-opaque-type-in-struct.slang [hunk 2]
```diff
@@ -4,7 +4,7 @@
 // aggregate type that includes an opaque type
 
 //TEST(compute):COMPARE_COMPUTE:
-//DISABLE_TEST(compute):COMPARE_COMPUTE:-slang -shaderobj -mtl
+//TEST(compute):COMPARE_COMPUTE:-slang -shaderobj -mtl
```
Comment (jkwak-work) [inline tests/language-feature/types/opaque/out-param-opaque-type-in-struct.slang:7]: This test appears to be failing,
```
FAILED test: 'tests/language-feature/types/opaque/out-param-opaque-type-in-struct.slang.1 (mtl)' 
EXPECTED{{{
12
23
44
85
}}}
ACTUAL{{{
0
0
0
0
}}}
```

File: tests/language-feature/types/opaque/return-opaque-type-in-struct.slang [hunk 3]
```diff
@@ -4,7 +4,7 @@
 // aggregate type that includes an opaque type
 
 //TEST(compute):COMPARE_COMPUTE:
-//DISABLE_TEST(compute):COMPARE_COMPUTE:-slang -shaderobj -mtl
+//TEST(compute):COMPARE_COMPUTE:-slang -shaderobj -mtl
```
Comment (jkwak-work) [inline tests/language-feature/types/opaque/return-opaque-type-in-struct.slang:7]: This test appears to be failing,
```
FAILED test: 'tests/language-feature/types/opaque/return-opaque-type-in-struct.slang.1 (mtl)' 
EXPECTED{{{
12
23
44
85
}}}
ACTUAL{{{
0
0
0
0
}}}
```

Comment (ArielG-NV) [reply tests/language-feature/types/opaque/return-opaque-type-in-struct.slang:7]: This PR is still being worked on, I thought it was working, but GFX CI was failing due to an infinite loop I did not test for.

**Review by kaizhangNV — APPROVED** [review]

**Pull request merged**